### PR TITLE
Save one query in Elements’ `_elementById()` when `$elementType` is `null`

### PR DESCRIPTION
**Pull request by benface**
Except when running Craft < 3.2.6.

### CODE REVIEW

**Comment by brandonkelly** [comment]
What is the context for this? Can you describe what the issue is and what you’re trying to improve?

**Comment by benface** [comment]
@brandonkelly I was just looking at this code to understand something about Craft's internals and I thought I saw an optimization opportunity, but maybe it's actually not? I needed to get an element by its id, but it can be any type, so I was under the impression that this would result in one less database query in that scenario.

File: src/services/Elements.php [hunk 1]
```diff
@@ -494,16 +494,29 @@ private function _elementById(string $property, $elementId, string $elementType
         if (!$elementId) {
             return null;
         }
-
-        if ($elementType === null) {
-            $elementType = $this->_elementTypeById($property, $elementId);
```
Comment (benface) [inline src/services/Elements.php:499]: First query.

File: src/services/Elements.php [hunk 2]
```diff
@@ -513,16 +526,6 @@ private function _elementById(string $property, $elementId, string $elementType
         $query->anyStatus();
 
         // Is this a draft/revision?
-        try {
-            $data = (new Query())
-                ->select(['draftId', 'revisionId'])
-                ->from([Table::ELEMENTS])
-                ->where([$property => $elementId])
-                ->one();
```
Comment (benface) [inline src/services/Elements.php:521]: Second query.

File: src/services/Elements.php [hunk 3]
```diff
@@ -494,16 +494,29 @@ private function _elementById(string $property, $elementId, string $elementType
...
-            $elementType = $this->_elementTypeById($property, $elementId);
-
-            if ($elementType === null) {
+        
+        try {
+            $data = (new Query())
+                ->select(['draftId', 'revisionId', 'type'])
+                ->from([Table::ELEMENTS])
+                ->where([$property => $elementId])
+                ->one();
```
Comment (benface) [inline src/services/Elements.php:503]: Now just one query that handles both things.

**Comment by brandonkelly** [comment]
Ahh, gotcha. Makes sense. Thanks for the PR!

**Comment by brandonkelly** [comment]
Craft 3.6.13 is out with this change.

**Pull request merged**